torchgen/model.py

```diff
@@ -1176,6 +1335,31 @@ def strip_ret_annotation(r: Return) -> Return:
         if strip_view_copy_name and base_name.endswith("_copy"):
             base_name = base_name.replace("_copy", "")
 
+        # find mutable inputs that are not originally returned, and convert them to returns
+        returns_from_mutable_inputs = tuple(
+            # When we're grouping functions we strip the return names,
+            # but when we're generating the actual functional variants then we follow
+            # a convention for what to name the returns
+            Return(name=f'{a.name}_out' if keep_return_names else None, type=a.type, annotation=None)
+            for a in itertools.chain(
+                # Order is important here (otherwise e.g. inplace with mutable args + out= with mutable args won't have the same signature)
+                [self.arguments.self_arg.argument] if self.arguments.self_arg is not None else [],
+                self.arguments.out,
+                self.arguments.post_self_positional,
+            )
+            if a.annotation is not None
+            and a.annotation.is_write
+            and not any(a.annotation == r.annotation for r in self.returns)
+        )
+        original_returns = tuple(map(strip_ret_annotation, self.returns))
+        # Ordering is important here. We expect the "mutable input" returns to come last.
+        returns = original_returns + returns_from_mutable_inputs
+
+        args_sig = self.arguments.signature(strip_default=strip_default)
+        # See Note [arange.start_step schema]
+        if str(self.name) == 'arange.start_step':
+            args_sig = Arguments.parse(str(args_sig).replace('Scalar step', 'Scalar step=1'))
```

Comment:
`arange.start_step/start_out` is actually one of the 6 mutable ops today that is "broken" (doesn't get grouped up by default because of a schema difference).

But I wanted to ensure that it works with functionalization directly in this PR, because functionalization already provides support for it.

Without being BC-breaking, I added this special case to ensure that it gets grouped properly. Alternatively I could probably have manually written a functionalization kernel for it, but getting it to group properly also has the nice benefit of letting LTC/XLA 100% rely on functional ops in their codegen (which let me simplify a bunch of other things).

Some time later, we can probably just fix this by (BC-breaking) changing the chsmea so that `step` is defaulted. We'll probably have to append the overload name to the C++ API name or something.